### ISSUE TITLE
class_delegate_protocol false positive

### ISSUE THREAD

**Issue by k4ety** [realm/SwiftLint#3311]
### New Issue Checklist

- [X] Updated SwiftLint to the latest version
- [X] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

The rule **class_delegate_protocol** currently allows a protocol to inherit a protocol ending with "Delegate" but fails when it inherits from a protocol ending in "DelegateType" or any other form containing, but not ending in, "Delegate".  I propose that the rule be changed to allow inheriting from any protocol with a name containing "Delegate".

```swift
// This triggers a violation:
protocol ModuleDelegateType: class {}
protocol ChildModuleDelegate: ModuleDelegateType {}

```

**Comment by jpsim** [comment]
Yup this is a clear limitation of the simple way in which the rule was designed.
